cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -122,6 +133,50 @@ type ClusterConfiguration struct {
 	ClusterName string `json:"clusterName,omitempty"`
 }
 
+// DNSAddOnType defines string identifying DNS add-on types
+type DNSAddOnType string
+
+const (
+	// CoreDNS add-on type
+	CoreDNS DNSAddOnType = "CoreDNS"
+
+	// KubeDNS add-on type
+	KubeDNS DNSAddOnType = "kube-dns"
+)
+
+// DNS defines the DNS addon that should be used in the cluster
+type DNS struct {
+	// Type defines string DNS add-on to be used
+	Type DNSAddOnType `json:"type"`
+
+	// ImageRepository, if set this fields allows to override the default image repository
+	// defined in ClusterConfiguration
+	ImageRepository string `json:"imageRepository,omitempty"`
```

Comment:
Not sure of what will be the use case of setting this to a different value (than `ClusterConfiguration.ImageRepository`, but not setting `Image` here directly).

Reply:
by setting ImageRepository you can do kubeadm upgrade, while image binds you to an image:tag